test/distributed/test_c10d.py

```diff
@@ -2576,20 +2576,26 @@ def check_no_grads():
         # No parameter should have their gradient set.
         check_no_grads()
 
-    @requires_nccl()
-    @skip_if_not_multigpu
-    def test_accumulate_gradients_no_sync(self):
-        # This is the recommended way to implement accumulate grads
+    def _test_accumulate_gradients_no_sync(self, num_iters=2, ddp_comm_hook=None):
+        """
+        This is the recommended way to implement accumulate grads.
+        If ``ddp_comm_hook`` input was specified, it will also register that hook
+        to the ``ddp_model``. The hook fed into this function should not change
+        the resulting gradients.
+        """
         int_devices = gpus_for_rank(self.world_size)[self.rank][:1]
-        devices = list([torch.device('cuda:' + str(i)) for i in int_devices])
+        devices = list([torch.device("cuda:" + str(i)) for i in int_devices])
```

Comment:
nit: since you're already changing this line, might as well remove the call to `list()` which is not needed

Reply:
Oh, I missed that. I was just running linter for the parts that I was changing. I also made relevant changes across the whole test file.